src/ResXManager.Model/ProjectFileExtensions.cs

```diff
@@ -15,12 +15,59 @@ public static class ProjectFileExtensions
         private static readonly string[] _supportedFileExtensions = { Resx, Resw };
 
         public static string GetBaseDirectory(this ProjectFile projectFile)
+        {
+            var extension = projectFile.Extension;
+            var filePath = projectFile.FilePath;
+            var directoryName = Path.GetDirectoryName(Resw.Equals(extension, StringComparison.OrdinalIgnoreCase) ? Path.GetDirectoryName(filePath) : filePath);
+
+            directoryName = projectFile.GetBaseLanguageDirectory();
+
+            return directoryName ?? throw new InvalidOperationException();
+        }
+
+        /// <summary>
+        /// Returns a subdirectory path relative to the language path.
+        /// Example:
+        /// D:\\myProject\\strings\\en-US\\subdir\\myview.resw
+        /// </summary>
+        /// <param name="projectFile"></param>
+        /// <returns></returns>
+        public static string GetSubDirectory(this ProjectFile projectFile)
+        {
+            var extension = projectFile.Extension;
+            var filePath = projectFile.FilePath;
+
+            var directoryName = Path.GetDirectoryName(filePath);
+            var languageDirectoryName = projectFile.GetBaseLanguageDirectory();
+            directoryName = directoryName.TrimEnd(Path.DirectorySeparatorChar);
+            languageDirectoryName = languageDirectoryName.TrimEnd(Path.DirectorySeparatorChar);
+
+            directoryName = directoryName.Replace(Path.Combine(languageDirectoryName, GetLanguageName(filePath)),"");
+
+            return directoryName;
+        }
+        /// <summary>
+        /// Please only use for resw files
+        /// Get the directory that contains language directories like "en-US" 
+        /// </summary>
+        /// <param name="projectFile"></param>
+        /// <returns></returns>
+        /// <exception cref="InvalidOperationException"></exception>
+        public static string GetBaseLanguageDirectory(this ProjectFile projectFile)
         {
             var extension = projectFile.Extension;
             var filePath = projectFile.FilePath;
 
             var directoryName = Path.GetDirectoryName(Resw.Equals(extension, StringComparison.OrdinalIgnoreCase) ? Path.GetDirectoryName(filePath) : filePath);
 
+            String directory = Path.GetDirectoryName(filePath);
+            String? languageName = Path.GetDirectoryName(filePath).Split(Path.DirectorySeparatorChar).FirstOrDefault(item => CultureHelper.IsValidCultureName(item));

```

Comment:
Calling this for *EVERY* file (even *.cs) in a real project slows down loading dramatically. When running in debugger, it takes several minutes to load a project.